creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -555,83 +406,35 @@ class SnapbackSM {
-      }
-
-      /**
-       * Select chunk of users to process in this run
-       *  - User is selected if (user_id % ModuloBase = currentModuloSlice)
-       *  - nodeUsers = array of objects of schema { primary, secondary1, secondary2, user_id, wallet }
-       */
-      nodeUsers = nodeUsers.filter(
-        nodeUser => (nodeUser.user_id % ModuloBase === this.currentModuloSlice)
-      )
-      decisionTree.push({
-        stage: '3/ select nodeUsers modulo slice',
-        vals: { nodeUsersSubsetLength: nodeUsers.length, moduloBase: ModuloBase, currentModuloSlice: this.currentModuloSlice },
-        time: Date.now()
-      })
-
-      // Compute content node peerset from nodeUsers (all nodes that are in a shared replica set with this node)
-      let peerSet = []
-      try {
-        peerSet = this.computeContentNodePeerSet(nodeUsers)
+      nodeUsers = await this.peerSetManager.getNodeUsers()
+      nodeUsers = this.peerSetManager.sliceUsers({ nodeUsers, moduloBase: ModuloBase, currentModuloSlice: this.currentModuloSlice })
 
-        decisionTree.push({
-          stage: '4/ computeContentNodePeerSet() Success',
-          vals: { peerSetLength: peerSet.size },
-          time: Date.now()
-        })
-      } catch (e) {
-        decisionTree.push({ stage: '4/ computeContentNodePeerSet() Error', vals: e.message, time: Date.now() })
-        throw new Error('processStateMachineOperation():computeContentNodePeerSet() Error')
-      }
-
-      /**
-       * Determine health for every peer & build list of unhealthy peers
-       * TODO change from sequential to chunked parallel
-       */
-      const unhealthyPeers = new Set()
-      try {
-        for await (const peer of peerSet) {
-          const peerIsHealthy = await this.determinePeerHealth(peer)
-          if (!peerIsHealthy) {
-            unhealthyPeers.add(peer)
-          }
-        }
+      decisionTree.push({ stage: 'getNodeUsers() and sliceUsers() Success', vals: { nodeUsersLength: nodeUsers.length }, time: Date.now() })
+    } catch (e) {
+      decisionTree.push({ stage: 'getNodeUsers() or sliceUsers() Error', vals: e.message, time: Date.now() })
+      throw new Error(`processStateMachineOperation():getNodeUsers()/sliceUsers() Error: ${e.toString()}`)
+    }
 
-        decisionTree.push({
-          stage: '5/ determinePeerHealth() Success',
-          vals: {
-            unhealthyPeerSetLength: unhealthyPeers.size,
-            healthyPeerSetLength: peerSet.size - unhealthyPeers.size,
-            unhealthyPeers: Array.from(unhealthyPeers)
-          },
-          time: Date.now()
-        })
-      } catch (e) {
-        decisionTree.push({ stage: '5/ determinePeerHealth() Error', vals: e.message, time: Date.now() })
-        throw new Error('processStateMachineOperation():determinePeerHealth() Error')
-      }
+    let unhealthyPeers = await this.peerSetManager.getUnhealthyPeers(nodeUsers)
```

Comment:
also looks like there's no error handling for this? if `getUnhealthyPeers()` throws we will never see decision tree error state log right

Reply:
this method is wrapped in a larger catch block that adds another error state, and logs the resulting decision tree https://github.com/AudiusProject/audius-protocol/pull/1502/files/0ab1cea70368be1e941dcb2efcf191d3788d4950#diff-62e7d629775fe1c67406f64ab0a79b6aa6ed5bfb98fe5ed4e2048dc45bf4ed88R593

Reply:
my bad, this is not correct. moving the peerSetManager calls into the global try/catch

Reply:
nice, thanks for fixing!

Reply:
yeah saw you reverted that, thanks for doing
but doesn't address my above comment - if `peerSetManager.getUnhealthyPeers()` call here fails, we will never call `decisionTree.push()` with an error state

Reply:
if line 427 ever fails, the larger try catch block will catch the error, and push that error into the decision tree. https://github.com/AudiusProject/audius-protocol/pull/1502/files/0ab1cea70368be1e941dcb2efcf191d3788d4950#diff-62e7d629775fe1c67406f64ab0a79b6aa6ed5bfb98fe5ed4e2048dc45bf4ed88L790-L800

i do think these explicit try/catch blocks in this flow are not necessary if we have a larger try/catch that catches any error thrown. i rather we use a `phases` decision tree logging (as we do on signup) to simplify this logic.

for the sake of this PR and to fit the rest of the structure of this general sync operation, I can go ahead and wrap this specific code block in a try/catch.

Reply:
if you haven't already i think its fine

the reason i had done the individual try-catch blocks was so i could push more intelligent error msgs, e.g. lines 535
```
      } catch (e) {
        decisionTree.push({
          stage: 'issueSyncRequests() Error',
          vals: {
            numSyncRequestsRequired,
            numSyncRequestsIssued,
            numSyncRequestErrors: (syncRequestErrors ? syncRequestErrors.length : null),
            syncRequestErrors
          },
          time: Date.now()
        })
        throw new Error('processStateMachineOperation():issueSyncRequests() Error')
```

Reply:
just for consistency, wrapping this call with a try/catch